File: Java Implementation/Sorting/Insertion Sort/InsertionSort.java
import java.util.*;
public class InsertionSort
{
	public void Insertion(int a[])
	{
		int i=0,key,n=a.length;
		for(int j=1;j<n;j++)
		{
			key=a[j];
			i=j-1;
			while(i>=0 && a[i]>key)
			{
				a[i+1]=a[i];
				i--;
			}
			a[i+1]=key;
		}
	
		for(int k=0;k<n;k++)
		{
			System.out.print(a[k]+"   ");
		}
	
	}

	public static void main(String[] args)
	{
		Scanner sc = new Scanner(System.in);
		InsertionSort is=new InsertionSort();
		
		System.out.println("Enter Size of array");
		
		int len=sc.nextInt();
		int arr[]=new int[len];
		
		for(int i=0;i<len;i++)
		{
			arr[i]=sc.nextInt();
		}
		is.Insertion(arr);
	}

}

File: Python Implementation/Sorting/InsertionSort.py
#Insertion Sort Implementation
n=int(input('Enter Number of elements in array'))

arr=list()

for i in range(0,n):
    arr.append(int(input()))

print('your given set of elements:',arr)

def insertion_sort(l:list):
    for i in range(1,len(l)):
        key=l[i]
        j=i-1
        while(j>0 and l[j]>key):
            l[j+1]=l[j]
            j=j-1
        l[j+1]=key
    return l

print(insertion_sort(arr))


